index.js

```diff
@@ -13,9 +13,12 @@ const config = require('./config');
 config.init();
 const plugins = require('./plugins');
 
+const windowSet = new Set([]);
+
 // expose to plugins
 app.config = config;
 app.plugins = plugins;
+app.getWindows = () => [...windowSet];
```

Comment:
(But I feel like `Array` is the lowest common denominator that gets the job done. Most people will be more familiar with that than a `Set`. If they really want, they can easily make it a `Set` themselves in plugin-land.)